xarray/tests/test_groupby.py

```diff
@@ -1411,14 +1420,14 @@ def center(x):
         expected_centered = expected_ds["foo"]
         assert_allclose(expected_centered, grouped.map(center))
 
-    def test_groupby_map_ndarray(self):
+    def test_groupby_map_ndarray(self) -> None:
         # regression test for #326
         array = self.da
         grouped = array.groupby("abc")
-        actual = grouped.map(np.asarray)
+        actual = grouped.map(np.asarray)  # type: ignore[arg-type] # TODO: Not sure using np.asarray like this makes sense with array api
```

Comment:
I think mypy is correct to complain here.

Maybe it's this particular example but `xp.asarray` is always converted to a xp array according to array api, haven't tried it myself but probably numpy 2.0 as well.

Reply:
I bet we need to adapt to the new apis here.